diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2296a0e41..bb16496d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,28 +35,10 @@ include(gen/sources.cmake)
 enable_language(C)
 enable_language(CXX)
 
-# TODO(crbug,com/389897612): Register tests with CTest. For now, we include this
-# only to pick up the standard |BUILD_TESTING| option.
-include(CTest)
 include(GNUInstallDirs)
 
 set(INSTALL_ENABLED 1)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND NOT CMAKE_CROSSCOMPILING AND
-   BUILD_TESTING)
-  find_package(PkgConfig QUIET)
-  if (PkgConfig_FOUND)
-    pkg_check_modules(LIBUNWIND libunwind-generic>=1.3.0)
-    if(LIBUNWIND_FOUND)
-      add_definitions(-DBORINGSSL_HAVE_LIBUNWIND)
-    else()
-      message("libunwind not found. Disabling unwind tests.")
-    endif()
-  else()
-    message("pkgconfig not found. Disabling unwind tests.")
-  endif()
-endif()
-
 string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LOWER)
 if(NOT FIPS)
   if(CMAKE_BUILD_TYPE_LOWER STREQUAL "relwithassert" OR
@@ -135,22 +117,6 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_XOPEN_SOURCE=700")
 endif()
 
-if(FUZZ)
-  if(NOT CLANG)
-    message(FATAL_ERROR "You need to build with Clang for fuzzing to work")
-  endif()
-
-  if(CMAKE_C_COMPILER_VERSION VERSION_LESS "6.0.0")
-    message(FATAL_ERROR "You need Clang ≥ 6.0.0")
-  endif()
-
-  add_definitions(-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION)
-  set(RUNNER_ARGS "-deterministic" "-fuzzer" "-shim-config" "fuzzer_mode.json")
-
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address,fuzzer-no-link -fsanitize-coverage=edge,indirect-calls")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address,fuzzer-no-link -fsanitize-coverage=edge,indirect-calls")
-endif()
-
 if(BUILD_SHARED_LIBS)
   add_definitions(-DBORINGSSL_SHARED_LIBRARY)
   # Enable position-independent code globally. This is needed because
@@ -377,59 +343,6 @@ if(USE_CUSTOM_LIBCXX)
          FILE ${PROJECT_BINARY_DIR}/bssl_test_only.cmake)
 endif()
 
-if(BUILD_TESTING)
-  # Add minimal googletest targets. The provided one has many side-effects, and
-  # googletest has a very straightforward build.
-  add_library(
-    boringssl_gtest
-    STATIC
-    third_party/googletest/googlemock/src/gmock-all.cc
-    third_party/googletest/googletest/src/gtest-all.cc
-  )
-  target_include_directories(
-      boringssl_gtest
-      PUBLIC
-      third_party/googletest/googlemock/include
-      third_party/googletest/googletest/include
-      PRIVATE
-      third_party/googletest/googlemock
-      third_party/googletest/googletest
-  )
-
-  # Wrap this in a function for variable scoping. Switch to block() when our
-  # minimum CMake is at least CMake 3.25.
-  function(add_google_benchmark)
-    set(GOOGLETEST_PATH "${PROJECT_SOURCE_DIR}/third_party/googletest")
-    set(BENCHMARK_DOWNLOAD_DEPENDENCIES OFF)
-    set(BENCHMARK_ENABLE_TESTING OFF)
-    set(BENCHMARK_ENABLE_INSTALL OFF)
-    set(BENCHMARK_INSTALL_DOCS OFF)
-    if(USE_CUSTOM_LIBCXX)
-      # benchmark's feature testing does not pick up the custom STL.
-      set(HAVE_STD_REGEX 1)
-    endif()
-    if(DEFINED CMAKE_OSX_SYSROOT)
-      # CMAKE_CROSSCOMPILING is not correctly set when using CMAKE_OSX_SYSROOT
-      # to specify iOS vs macOS. Set CMAKE_CROSSCOMPILING so benchmark feature
-      # detection does not try to run its test programs.
-      set(CMAKE_CROSSCOMPILING 1)
-    endif()
-    add_subdirectory(third_party/benchmark)
-  endfunction()
-  add_google_benchmark()
-endif()
-
-if(FUZZ AND LIBFUZZER_FROM_DEPS)
-  file(GLOB LIBFUZZER_SOURCES "util/bot/libFuzzer/*.cpp")
-  add_library(Fuzzer STATIC ${LIBFUZZER_SOURCES})
-  # libFuzzer must be built without -fsanitize-coverage options or clang
-  # crashes.
-  set_target_properties(
-    Fuzzer PROPERTIES
-    COMPILE_FLAGS "-fsanitize-coverage=0"
-  )
-endif()
-
 # Enable a suite of warnings. We do this here to avoid applying to external
 # dependencies, built above. Note this uses the list-based add_compile_options,
 # instead of CMAKE_*_FLAGS (string-based). Changes to CMAKE_*_FLAGS still apply
@@ -702,77 +615,9 @@ if(MSVC)
     "$<$<COMPILE_LANGUAGE:C,CXX>:${PKI_MSVC_DISABLED_WARNINGS_LIST}>")
 endif()
 
-if(BUILD_TESTING)
-  add_library(test_support_lib STATIC
-              ${TEST_SUPPORT_SOURCES} ${TEST_SUPPORT_SOURCES_ASM_USED})
-  if(LIBUNWIND_FOUND)
-    target_compile_options(test_support_lib PRIVATE ${LIBUNWIND_CFLAGS_OTHER})
-    target_include_directories(test_support_lib PRIVATE ${LIBUNWIND_INCLUDE_DIRS})
-    target_link_libraries(test_support_lib ${LIBUNWIND_LDFLAGS})
-  endif()
-  if(WIN32)
-    target_link_libraries(test_support_lib dbghelp)
-  endif()
-  target_link_libraries(test_support_lib boringssl_gtest crypto)
-
-  # Declare a dummy target to build all unit tests. Test targets should inject
-  # themselves as dependencies next to the target definition.
-  add_custom_target(boringssl_all_tests)
-
-  # TODO(crbug.com/42290412): Drive this build with build.json and flatten.
-  add_subdirectory(ssl/test)
-
-  # urandom_test is a separate binary because it needs to be able to observe the
-  # PRNG initialisation, which means that it can't have other tests running before
-  # it does.
-  add_executable(urandom_test ${URANDOM_TEST_SOURCES})
-  target_link_libraries(urandom_test test_support_lib boringssl_gtest crypto)
-  add_dependencies(boringssl_all_tests urandom_test)
-
-  add_executable(crypto_test ${CRYPTO_TEST_SOURCES})
-  target_link_libraries(crypto_test test_support_lib boringssl_gtest crypto)
-  add_dependencies(boringssl_all_tests crypto_test)
-
-  add_executable(ssl_test ${SSL_TEST_SOURCES})
-  target_link_libraries(ssl_test test_support_lib boringssl_gtest ssl crypto)
-  add_dependencies(boringssl_all_tests ssl_test)
-  add_executable(decrepit_test ${DECREPIT_TEST_SOURCES})
-  target_link_libraries(decrepit_test test_support_lib boringssl_gtest
-                        decrepit ssl crypto)
-  add_dependencies(boringssl_all_tests decrepit_test)
-
-  add_executable(pki_test ${PKI_TEST_SOURCES})
-  target_link_libraries(pki_test test_support_lib boringssl_gtest pki crypto)
-  target_compile_options(pki_test PRIVATE ${PKI_CXX_FLAGS})
-  add_dependencies(boringssl_all_tests pki_test)
-
-  add_executable(test_fips util/fipstools/test_fips.cc)
-  target_link_libraries(test_fips crypto)
-
-  add_executable(jitter_deltas util/fipstools/entropy/jitter_deltas.cc)
-  target_link_libraries(jitter_deltas crypto)
-
-  if(BUILD_TESTING)
-    add_executable(bssl_bench ${BENCH_SOURCES})
-    target_link_libraries(bssl_bench ssl pki crypto benchmark::benchmark)
-  endif()
-endif()
-
-if(FIPS)
-  add_executable(modulewrapper ${MODULEWRAPPER_SOURCES})
-  target_link_libraries(modulewrapper crypto)
-  add_executable(entropy_modulewrapper ${ENTROPY_MODULEWRAPPER_SOURCES})
-  target_link_libraries(entropy_modulewrapper crypto)
-endif()
-
 add_executable(bssl ${BSSL_SOURCES})
 target_link_libraries(bssl ssl crypto)
 
-if(FUZZ)
-  # TODO(crbug.com/42290412): Drive this build with build.json and flatten.
-  add_subdirectory(fuzz)
-endif()
-
 if(RUST_BINDINGS)
   find_program(BINDGEN_EXECUTABLE bindgen REQUIRED)
   add_subdirectory(rust)
@@ -782,68 +627,6 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set(HANDSHAKER_ARGS "-handshaker-path" $<TARGET_FILE:handshaker>)
 endif()
 
-if(BUILD_TESTING)
-  if(FIPS)
-    add_custom_target(
-      acvp_tests
-      COMMAND ${GO_EXECUTABLE} build -o ${CMAKE_CURRENT_BINARY_DIR}/acvptool
-              boringssl.googlesource.com/boringssl.git/util/fipstools/acvp/acvptool
-      COMMAND ${GO_EXECUTABLE} build -o ${CMAKE_CURRENT_BINARY_DIR}/testmodulewrapper
-              boringssl.googlesource.com/boringssl.git/util/fipstools/acvp/acvptool/testmodulewrapper
-      COMMAND cd util/fipstools/acvp/acvptool/test &&
-              ${GO_EXECUTABLE} run check_expected.go
-              -tool ${CMAKE_CURRENT_BINARY_DIR}/acvptool
-              -module-wrappers modulewrapper:$<TARGET_FILE:modulewrapper>,testmodulewrapper:${CMAKE_CURRENT_BINARY_DIR}/testmodulewrapper
-              -tests tests.json
-      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-      DEPENDS modulewrapper
-      USES_TERMINAL)
-
-    add_custom_target(
-      fips_specific_tests_if_any
-      DEPENDS acvp_tests
-    )
-  else()
-    add_custom_target(fips_specific_tests_if_any)
-  endif()
-
-  file(STRINGS util/go_tests.txt GO_TESTS)
-  set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS
-              util/go_tests.txt)
-
-  # CMake target names can clash when projects are imported with
-  # add_subdirectory. Only emit the run_tests target if we're the top-level
-  # target. See https://crbug.com/428112180.
-  #
-  # Also disable on Visual Studio for now due to name clash.
-  # See https://crbug.com/465077378.
-  if(PROJECT_IS_TOP_LEVEL AND NOT CMAKE_GENERATOR MATCHES "^Visual Studio ")
-    if(GO_EXECUTABLE)
-      add_custom_target(
-          run_tests
-          COMMAND ${CMAKE_COMMAND} -E echo "Running Go tests"
-          COMMAND ${GO_EXECUTABLE} test ${GO_TESTS}
-          COMMAND ${CMAKE_COMMAND} -E echo
-          COMMAND ${CMAKE_COMMAND} -E echo "Running unit tests"
-          COMMAND ${GO_EXECUTABLE} run util/all_tests.go -build-dir
-                  ${CMAKE_CURRENT_BINARY_DIR}
-          COMMAND ${CMAKE_COMMAND} -E echo
-          COMMAND ${CMAKE_COMMAND} -E echo "Running SSL tests"
-          COMMAND cd ssl/test/runner &&
-                  ${GO_EXECUTABLE} test -shim-path $<TARGET_FILE:bssl_shim>
-                    ${HANDSHAKER_ARGS} ${RUNNER_ARGS}
-          WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-          DEPENDS boringssl_all_tests bssl_shim handshaker fips_specific_tests_if_any
-          USES_TERMINAL)
-    else()
-      add_custom_target(
-          run_tests
-          COMMAND ${CMAKE_COMMAND} -E echo "Running tests requires Go"
-          COMMAND ${CMAKE_COMMAND} -E false)
-    endif()
-  endif()
-endif()
-
 if(BORINGSSL_PREFIX)
   require_go()
   add_custom_target(
